src/Behat/Behat/EventDispatcher/Cli/StopOnFailureController.php

```diff
@@ -36,24 +36,18 @@
 final class StopOnFailureController implements Controller
 {
     /**
-     * @var EventDispatcherInterface
+     * @var StopOnFailureHandler
      */
-    private $eventDispatcher;
-
-    /**
-     * @var ResultInterpretation
-     */
-    private $resultInterpretation;
+    private $stopOnFailureHandler;
 
     /**
      * Initializes controller.
      *
-     * @param EventDispatcherInterface $eventDispatcher
+     * @param StopOnFailureHandler $stopOnFailureHandler
      */
-    public function __construct(EventDispatcherInterface $eventDispatcher)
+    public function __construct(StopOnFailureHandler $stopOnFailureHandler)
```

Comment:
This looks like a BC break, can it be avoided? e.g. by still passing in the dispatcher & creating the StopOnFailureHandler in the constructor?